ls1intum/Artemis
# Set exercise into task extension

### Checklist

### Motivation and Context
Bugfix

### Description
ProgrammingExerciseInstructionComponent is used to display instructions in multiple parts of the application. While we use the exercise to uniquely identify tasks now (needed for multiple programming exercises in exams), we do not propagate the exercise properly to the task extension service yet in other use cases (e.g. tutor dashboard view exercise)
Additionally we don't refresh the result stored in the singleton task wrapper service, when we navigate to an course exercise which does not have a participation.

### Steps for Testing


1. Check every part of the application where instructions might be displayed (tutor-exercise dashoard, course-exercise-instructions, programming exercise-details, code-editor, edit programming exercise, edit instructions in instructor code editor)
--> Should be displayed and no console errors should be thrown
